modMcf/src/org/aion/mcf/db/ContractDetailsCacheImpl.java

```diff
@@ -49,7 +50,14 @@ public static ContractDetailsCacheImpl copy(ContractDetailsCacheImpl cache) {
      */
     @Override
     public void put(ByteArrayWrapper key, ByteArrayWrapper value) {
-        storage.put(key, value);
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(value);
+
+        if (value.isZero()) {
+            storage.remove(key);
```

Comment:
This was part of the put/delete separation that was delegated to a later commit. I worry it may cause consensus issues.

Reply:
I thought the separation pulled these two out into their own methods? What should be done here then?

Reply:
I think for now we can leave this method unchanged. The update for contract details will be added as a separate PR.

Reply:
will do